### PR TITLE
Add support for null format property of Hive tables

### PR DESCRIPTION
**Pull request by jirassimok**
Allow Presto to correctly read from exernal Hive tables that use `serialization.null.format` to sotre NULL in an alternate format.

This works with `TEXTFILE`, `RCTEXT`, and `SEQUENCEFILE` formats. There are no tests for the latter two (because they can't be easily embedded in the tests because they are binary file formats and Presto can't insert data into external Hive tables).

### CODE REVIEW

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 1]
```diff
@@ -3597,6 +3634,8 @@ public void testCreateTableWithInvalidProperties()
                 .hasMessageMatching("Cannot specify skip_header_line_count table property for storage format: ORC");
         assertThatThrownBy(() -> assertUpdate("CREATE TABLE test_orc_skip_footer (col1 bigint) WITH (format = 'ORC', skip_footer_line_count = 1)"))
                 .hasMessageMatching("Cannot specify skip_footer_line_count table property for storage format: ORC");
+        assertThatThrownBy(() -> assertUpdate("CREATE TABLE test_orc_skip_footer (col1 bigint) WITH (format = 'ORC', null_format = 1)"))
```
Comment (losipiuk) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java:3619]: Please pass string as value for parameter for statement to look valid.

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 2]
```diff
@@ -3251,115 +3252,96 @@ public void testShowCreateTable()
         assertEquals(getOnlyElement(actualResult.getOnlyColumnAsSet()), createTableSql);
     }
 
-    @Test
-    public void testCreateExternalTable()
+    /**
+     * Boilerplate wrapper for tests creating external textfile tables.
```
Comment (losipiuk) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java:3256]: We are rarely doing javadoc comments in the codebase as they tend to get out of sync with the code.
I would drop this one as method name and parameters are self-explanatory enough.

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 3]
```diff
@@ -3251,115 +3252,96 @@ public void testShowCreateTable()
...
-        assertQuery(format("SELECT action, name FROM %s", tableName), "VALUES ('hello', NULL), (NULL, 123)");
-        assertUpdate(format("DROP TABLE %s", tableName));
+    @Test
+    public void testCreateExternalTableWithFieldSeparatorEscape() throws Exception
+    {
+        testCreateExternalTable(
+                "test_create_external_text_file_with_field_separator_and_escape",
+                "HelloEFFWorld\nByeEFFWorld",
+                "VALUES ('HelloF', 'World'), ('ByeF', 'World')",
+                // Properties
```
Comment (losipiuk) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java:3331]: drop comment

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 4]
```diff
@@ -3251,115 +3252,96 @@ public void testShowCreateTable()
         assertEquals(getOnlyElement(actualResult.getOnlyColumnAsSet()), createTableSql);
     }
 
-    @Test
-    public void testCreateExternalTable()
+    /**
```
Comment (losipiuk) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java:3255]: The `Also, add a separate test for the default field delimiter` should be extracted to separate commit.

**Comment by losipiuk** [comment]
```
2020-06-18T02:27:32.4934712Z [INFO] --- modernizer-maven-plugin:1.7.1:modernizer (modernizer) @ presto-hive ---
2020-06-18T02:27:32.5529715Z [ERROR] /home/runner/work/presto/presto/presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java:3422: Prefer String.toLowerCase(java.util.Locale)

```

File: presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java [hunk 5]
```diff
@@ -70,6 +70,7 @@
 import java.util.Map;
```
Comment (losipiuk) [inline presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java:70]: I have just notice that it would be nicer to move this refactoring commit before "Add null format property to Hive textfile tables".
And then only add `testCreateExternalTableWithNullFormat` in new form.
Now we are adding `testCreateExternalTableWithNullFormat` and then refactoring it in followup commit.

It is nice to have. But not a must do.

**Comment by losipiuk** [comment]
I have second thoughts about the name of parameter. If `null_value` would not be better than `null_format`. @electrum, @findepi opinion on that?

**Comment by findepi** [comment]
> This works with TEXTFILE, RCTEXT, and SEQUENCEFILE formats. 

@jirassimok what did we end up doing with RCTEXT and SEQUENCEFILE?


>  I have second thoughts about the name of parameter. If null_value would not be better than null_format. opinion on that?

@losipiuk 

Since this is applied to TEXTFILE only, it should probably be `textfile_....`, unless the plan is to extend it to RCTEXT and SEQUENCEFILE.

If i were to come with my own name, i would use `..._null_representation`, but there is strong advantage to using name resembling Hive's equivalent.
Hence i propose `textfile_null_format` or `null_format` (depending on the RCTEXT/SEQUENCEFILE).

**Comment by losipiuk** [comment]
> > This works with TEXTFILE, RCTEXT, and SEQUENCEFILE formats.
> 
> @jirassimok what did we end up doing with RCTEXT and SEQUENCEFILE?

We decided to have followup PR with that. That one will also include product tests. Jacob is working on that.

> 
> > I have second thoughts about the name of parameter. If null_value would not be better than null_format. opinion on that?
> 
> @losipiuk
> 
> Since this is applied to TEXTFILE only, it should probably be `textfile_....`, unless the plan is to extend it to RCTEXT and SEQUENCEFILE.

We plan.

> 
> If i were to come with my own name, i would use `..._null_representation`, but there is strong advantage to using name resembling Hive's equivalent.
> Hence i propose `textfile_null_format` or `null_format` (depending on the RCTEXT/SEQUENCEFILE).

We are generally not very consistent with hive if you look at the names we have already (see `textfile_field_separator` vs `field.delim` for example). 
I do not have a very strong feeling if we should use one name or another. Just wanted to run in through you, as you may have :) Let's keep it `null_format` then.
